src/shortestpaths/dijkstra.jl

```diff
@@ -47,6 +47,7 @@ function dijkstra_shortest_paths(
     dists[srcs] = zero(T)
     pathcounts[srcs] = 1
     closest_vertices = Vector{U}()  # Maintains vertices in order of distances from source
+    popped_out = Vector{Bool}(zeros(nvg))   # popped_out[u]=true if shortest distance from src to u is found
```

Comment:
OK, what if we used a `PriorityQueue` here that mapped vertex to distance? Then, when you added something to the queue, you'd be rewriting it if it already existed:
```
julia> q = PriorityQueue(Char, Int)

julia> q['a'] = 4
4

julia> q['b'] = 7
7

julia> q['c'] = 2
2

julia> q['a'] = 4
4

julia> q['b'] = 7
7

julia> q['c'] = 2
2

julia> q['b'] = 1
1

julia> dequeue!(q)
'b': ASCII/Unicode U+0062 (category Ll: Letter, lowercase)

julia> dequeue!(q)
'c': ASCII/Unicode U+0063 (category Ll: Letter, lowercase)

julia> dequeue!(q)
'a': ASCII/Unicode U+0061 (category Ll: Letter, lowercase)

julia> dequeue!(q)
ERROR: BoundsError: attempt to access 0-element Array{Pair{Char,Int64},1} at index [1]
```

@jpfairbanks - any memory usage disadvantages with this relative to using a Vector of Bool?